csrc/ir/utils.cpp

```diff
@@ -1310,4 +1310,111 @@ MmaOpDetails getMmaOpDetails(
   return details;
 }
 
+// Verifies the assumptions made when evaluating a fusion containing MmaOp:
+// 1. MmaOp is preceded by a broadcast.
+// 2. The inputs to MmaOp are broadcasted as the last dim for the first operand
+// and the first dim for the second operand. The inputs of MmaOp will be [M, K,
+// 1] x [1, K, N].
+void verifyMmaOpForEvaluation(MmaOp* mma_op, DataType final_out_dtype) {
+  const Val* in_a = mma_op->inA();
+  const Val* in_b = mma_op->inB();
+
+  NVF_ERROR(
+      in_a->as<TensorView>()->nDims() == in_b->as<TensorView>()->nDims(),
```

Comment:
`in_a/b->as<TensorView>` is used several times in this function. How about "caching" them (e.g. `in_a/b_tv = in_a/b->as<TensorView>`) before this line?